### DISCUSSION TITLE
SQLite

### DISCUSSION THREAD

**Discussion by syn74ks** [VladimirKalachikhin/Galadriel-cache#3] [General]
Hi, I thank you for providing this mapping tool - it is an excellent resource.

I am wondering if it would be difficult to modify your creation so it can deliver data from an SQLite file, rather than the tile-based system it currently works with.

I have no idea how much work is involved, but though I would ask.

I have a lot of data - around 3TB and managing that number of files is proving quite difficult (using Windows). Windows grinds to a halt when trying to manage that many files on a single drive, so I moved to the SQLite storage method.

Thanks again.

**Comment by VladimirKalachikhin** [comment]
Thank you for your kind words.

Yes, in Windows no way to store hundreds of thousands of files in the native to Windows file system. But in Linux there is no problem with this. The file systems used in Linux allow you to do this.
The reason to store tiles in x/y/z file structure is to make map available to other application directly. From OruxMaps, for example. Or other. The SD card with ext4 file system may be mount on Android. In this way, the maps remain available on server hardware crash.
So not problem to store tiles in SQLite, but this do maps unavailable to other applications.

However, for non onboard usage, SQLite storage may be useful. If there is such a need, those who wish can make a donation for the development.

**Reply by syn74ks** [reply]
Hi Vladimir,

Many thanks for your response.

I expected that Linux would handle very large numbers of files without a problem.

The issue I face is the use of SASPlanet. This is the software I use to download and store map files, then export to the specific application. As far as I know, there is no Linux version of this software.

If you can recommend some other software, I would be happy to use it.

Though, SASPlanet will convert from the SQLite file format I currently use, to the x/y/z format - it will just take a while to do  :)

Thanks again,

David

**Reply by VladimirKalachikhin** [reply]
Heh.
SAS.Planet work fine on Wine.
But my choice for converting from x/y/z to other -- is [MOBAC](https://mobac.sourceforge.io/)

But see to this:
![cover map](https://user-images.githubusercontent.com/32364744/129491304-808b8f8e-4fc4-497e-99d5-4bbfa7a7a343.jpg)
This is a cover map. Each shaded pixel indicate a tile present.

**Reply by syn74ks** [reply]
Thanks - I forgot about WINE.

I will check out the mapping tool above.

What I like about SASPlanet is that the ESRI mapping is already there. I like to download satellite imagery most of the time.

The other issue with storing in the x/y/z format is the potential wasted space and file-system overheads. What allocation unit size have you found works best for the x/y/z storage format?

Thanks,

David

**Reply by VladimirKalachikhin** [reply]
About ESRI -- pay attention to https://github.com/VladimirKalachikhin/Galadriel-map/tree/master/emergencykit
About disk usage... See this: https://wiki.openstreetmap.org/wiki/Tile_disk_usage
I believe that the required storage volumes are not so large that take care of potential wasted space. 
I haven't done any research, and I format SD card as 4096 B/block and one i-node by block. May be, 1024 B/block and one i-node by block will be better.

**Reply by syn74ks** [reply]
Hi again,

Can you email me with some pricing information for the 4G router you recommend?

david.prnich@gmail.com

Thanks.

**Comment by syn74ks** [comment]
Hi again,

I tried to use your pre-made image for the Raspberry Pi to test out GaladrielMap and Cache but after uploading the image to the microSD and tried to boot, I got a kernel panic - it would not boot.

I have attached an image of the issue.

I was only using an 8GB microSD - will try again with a 32GB in a couple of days.


![IMG_20210821_232036](https://user-images.githubusercontent.com/18433583/130326557-e493b366-e5f6-4e48-bed7-0de7c98db2b6.jpg)

**Comment by VladimirKalachikhin** [comment]
You can't just burn 32GB image to 8GB flashcard.
You can mount the image and truncate the volume size to 8GB. By GParted, for example. But you will have little space left for tiles.
